torch/csrc/jit/codegen/cuda/grouped_reduction.cpp

```diff
@@ -57,6 +57,13 @@ void validateReductionGrouping(
   const auto num_root_dims = ref_domain.size();
   const auto num_dims = ref_tv->nDims();
   const auto ref_ca_pos = ref_tv->getComputeAtPosition();
+  const auto ref_cw_pos = ref_tv->getComputeWithPosition();
+  // Don't know which consumer would be computed with at this
+  // point. Just make sure all the grouped reduction outputs have the
+  // same set of consumers. This is not necessarily a required
+  // condition and could be made more flexible
```

Comment:
Right now reduction grouping is provided as a scheduling primitive (`groupReductions(std::vector<TensorView*)`) so that schedulers can opt-in to group particular reductions. This part of the code is part of the validations done before converting multiple `ReductionOp` exprs to a single `GroupedReductionOp` expr. 

Everything happens before lowering, so there can be unresolved computeWith. Specifically, what this validation checks is that if the output of one of the grouped `ReductionOp` exprs is set to be computed-with, all of the other reduction outputs must have non-conflicting compute-with settings. That means all of the outputs have the same computeWith position. At this point, they are unresolved, so we don't know which consumer of each output will be picked as the target of the computeWith, so I simply enforced all of them must have the same consumers, which is more than enough to make the computeWith setting after grouping still valid.